File: bench/map_iter.go
// Go peer of bench/map_iter.vader. `map[int32]int32` iterated with the
// built-in `for k, v := range m` loop — Go's most idiomatic shape and
// the closest analogue to Vader's `for entry in m`.

package main

import "fmt"

const (
	innerKeys  = 1_000
	outerLoops = 1_000
)

func main() {
	m := make(map[int32]int32, innerKeys)
	for i := int32(0); i < innerKeys; i++ {
		m[i] = i
	}
	var total int64
	for outer := 0; outer < outerLoops; outer++ {
		for _, v := range m {
			total += int64(v)
		}
	}
	fmt.Printf("map_iter inner=%d outer=%d total=%d\n", innerKeys, outerLoops, total)
}

File: bench/map_iter.ts
// TypeScript peer of bench/map_iter.vader. JS `Map` keyed by i32 ;
// repeated iteration via `for...of map`. Bun's V8 inlines the iterator
// protocol well — closest match to Vader's fused for-in dispatch path.

const INNER_KEYS = 1_000;
const OUTER_LOOPS = 1_000;

const m = new Map<number, number>();
for (let i = 0; i < INNER_KEYS; i++) m.set(i, i);

let total = 0n;
for (let outer = 0; outer < OUTER_LOOPS; outer++) {
  for (const [, value] of m) {
    total += BigInt(value);
  }
}
console.log(`map_iter inner=${INNER_KEYS} outer=${OUTER_LOOPS} total=${total}`);
